### PR TITLE
Include untriggered conditional orders in stateful order count

### PR DESCRIPTION
**Pull request by jayy04**
### Changelist
- Include untriggered conditional orders in stateful order count.

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

**Review by lucas-dydx — APPROVED** [review]
Should we add an error log if the stateful order count is zero [here](https://github.com/dydxprotocol/v4-chain/blob/1b29b33d01d3bfcc9dccfc692b9fe5c223c29698/protocol/x/clob/keeper/stateful_order_state.go#L155)?

File: protocol/x/clob/keeper/stateful_order_state.go [hunk 1]
```diff
@@ -152,7 +152,14 @@ func (k Keeper) DeleteLongTermOrderPlacement(
 	count := k.GetStatefulOrderCount(ctx, orderId.SubaccountId)
 	orderKey := orderId.ToStateKey()
 	if memStore.Has(orderKey) {
-		count--
+		if count == 0 {
+			k.Logger(ctx).Error(
+				"Stateful order count is zero but but order is in state. Underflow",
```
Comment (lucas-dydx) [inline protocol/x/clob/keeper/stateful_order_state.go:157]: Should we add the order ID in the log?

Comment (lucas-dydx) [inline protocol/x/clob/keeper/stateful_order_state.go:157]:
```suggestion
				"Stateful order count is zero but order is in the memstore. Underflow",
```

**Review by lucas-dydx — APPROVED** [review]

File: protocol/x/clob/keeper/orders_test.go [hunk 2]
```diff
@@ -2084,16 +2077,6 @@ func TestInitStatefulOrders(t *testing.T) {
 
 			// Run the test and verify expectations.
 			ks.ClobKeeper.InitStatefulOrders(ks.Ctx)
-			for subaccountId, count := range expectedStatefulOrderCounts {
```
Comment (lcwik) [inline protocol/x/clob/keeper/orders_test.go:2087]: keeping this check would have been nice since we could validate the count across more cases

**Pull request merged**